http/codegen/openapi/json_schema.go

```diff
@@ -271,7 +274,7 @@ func GenerateTypeDefinition(api *design.APIExpr, ut *design.UserTypeExpr) {
 }
 
 // TypeSchema produces the JSON schema corresponding to the given data type.
-func TypeSchema(api *design.APIExpr, t design.DataType) *Schema {
+func TypeSchema(api *design.APIExpr, t design.DataType, val *design.ValidationExpr) *Schema {
```

Comment:
In goa v2, validations are tied to an `AttributeExpr` and not a `DataType`. So moving the validation logic from `buildAttributeSchema` to here doesn't sound right. Would it be possible to explain why you moved the logic?